monitor/services/src/main/java/org/datacleaner/monitor/configuration/ComponentsCache.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.monitor.configuration;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Class ComponentsCache
+ * For caching and storing Components and configuration
+ *
+ * @author k.houzvicka
+ * @since 24.7.15
+ */
+public class ComponentsCache {
+    private static final Logger logger = LoggerFactory.getLogger(ComponentsCache.class);
+
+    ConcurrentHashMap<String, ComponentsCacheConfigWrapper> data = new ConcurrentHashMap<>();
```

Comment:
Maybe (can be future improvement) we should rather use a Guava cache here with a timeout. I wouldn't want it to be a short timeout or anything like that, but maybe after a day or two we should clean up just to prevent a long-running server instance from growing out of memory on ancient components.

Alternatively it could be a configuration item when creating the component. I guess it deserves it's own issue.